### PR TITLE
feat(notifications): Add Notification DTO and Model

### LINKED ISSUE #510: [V2 notifications] Create Notification DTO and Model
Create Notification DTO and Model for V2 API.

### PR DESCRIPTION
**Pull request by FelixTing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Added Notification DTO and Model for Support-Notifications V2 API.

Close #510 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [ ] No

### CODE REVIEW

File: v2/dtos/notification.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+)
+
+// Notification and its properties are defined in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/Notification
+type Notification struct {
+	common.Versionable `json:",inline"`
+	Id                 string   `json:"id,omitempty" validate:"omitempty,uuid"`
+	Created            int64    `json:"created,omitempty"`
+	Modified           int64    `json:"modified,omitempty"`
+	Category           string   `json:"category,omitempty" validate:"required_without=Labels,omitempty,oneof='SECURITY' 'SW_HEALTH' 'HW_HEALTH'"`
```
Comment (lenny-intel) [inline v2/dtos/notification.go:20]: @cloudxxx8 , @jpwhitemn do we need to rethink these Notification categories? They don't make sense to me. Why do we restrict them to these values?

File: v2/models/notification.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package models
+
+// Notification and its properties are defined in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/Notification
+// Model fields are same as the DTOs documented by this swagger. Exceptions, if any, are noted below.
+type Notification struct {
+	Timestamps
+	Category    Category
```
Comment (lenny-intel) [inline v2/models/notification.go:13]: Why is type `Category` and not `NotificationCategory` like `NotificationSeverity` & `NotificationStatus`

Comment (FelixTing) [reply v2/models/notification.go]: Renamed type `Category` to `NotificationCategory`.

**Review by lenny-intel — COMMENTED** [review]
LGTM, but still have open question on values for Category.

File: v2/dtos/notification.go (hunk 1 above)
Comment (cloudxxx8) [reply v2/dtos/notification.go:20]: OK, let's drop the hard code category in V2

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by FelixTing** [comment]
resolved merge conflicts

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
@cloudxxx8 , I notice the commits are not being squashed. We should always be using the `Squash and Merge` option rather than just `Merge` option for the Button.

image

**Comment by cloudxxx8** [comment]
@lenny-intel sorry, Jude reminded me before, but I keep forgetting it.
I will pay more attention in the future.

**Pull request merged**